### PR TITLE
Added public getters for EPH PartitionContext Offset and SequenceNumber

### PR DESCRIPTION
**Pull request by jtaubensee**

### CODE REVIEW

File: csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs [hunk 1]
```diff
@@ -22,6 +22,28 @@ internal PartitionContext(EventProcessorHost host, string partitionId, string ev
             this.ThisLock = new object();
         }
 
+        /// <summary>
+        /// Returns the current offset of the partition.
+        /// </summary>
+        public string Offset
```
Comment (hmlam) [inline csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs]: >public string Offset [](start = 8, length = 20)

just dog public string Offset { get; private set;}
Also if we are exposing this, then we should agree on that null is ok for Offset (currently I don't think the class initialize it)

File: csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs [hunk 2]
```diff
@@ -22,6 +22,28 @@ internal PartitionContext(EventProcessorHost host, string partitionId, string ev
...
+            get
+            {
+                return this.offset;
+            }
+        }
+
+        /// <summary>
+        /// Returns the current sequence number of the partition.
+        /// </summary>
+        public long SequenceNumber
```
Comment (hmlam) [inline csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs]: >SequenceNumber [](start = 20, length = 14)

Just do public long SequenceNumber { get; private set;}

**Comment by hmlam** [comment]
:clock1:

File: csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs [hunk 3]
```diff
@@ -20,29 +18,19 @@ internal PartitionContext(EventProcessorHost host, string partitionId, string ev
             this.EventHubPath = eventHubPath;
             this.ConsumerGroupName = consumerGroupName;
             this.ThisLock = new object();
+            this.SequenceNumber = 0;
+            this.Offset = PartitionReceiver.StartOfStream;
```
Comment (jtaubensee) [inline csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs]: initialized the SequenceNumber and Offset in internal the constructor. Offset constant should be -1. Technically you shouldnt receive null then, right?

File: csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs [hunk 4]
```diff
@@ -20,8 +18,20 @@ internal PartitionContext(EventProcessorHost host, string partitionId, string ev
...
         }
 
+        /// <summary>
+        /// Returns the current Offset of the partition.
```
Comment (serkantkaraca) [inline csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs]: >Returns the current Offset of the partition. [](start = 12, length = 44)

Can you mention "last checkpoint of partition"?

File: csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs [hunk 5]
```diff
@@ -20,8 +18,20 @@ internal PartitionContext(EventProcessorHost host, string partitionId, string ev
...
+        /// </summary>
+        public string Offset { get; private set; }
+
+        /// <summary>
+        /// Returns the current sequence number of the partition.
```
Comment (serkantkaraca) [inline csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs]: >/// Returns the current sequence number of the partition. [](start = 8, length = 57)

Can you mention "last checkpoint of partition"?

Comment (jtaubensee) [inline csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs]: so is the right wording "Returns the current sequence number from the last checkpoint of the partition"?

Comment (JamesBirdsall) [inline csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs]: Is that actually true? It's not for Java EPH.

Comment (serkantkaraca) [inline csharp/src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs]: James, any idea why we need "public void SetOffsetAndSequenceNumber(string offset, long sequenceNumber)"? This public method sets offset and sequence number w/o checkpoint. I thought whatever the PartitionContext is pointing that should be the same as checkpoint.

**Comment by jtaubensee** [comment]
Closing as we have decided not to publicly expose offset and sequence number.

**Pull request closed without merging**